components/mediators/bam/org.wso2.carbon.mediator.bam/src/main/java/org/wso2/carbon/mediator/bam/BamMediator.java

```diff
@@ -35,13 +38,48 @@
 
     private String serverProfile = "";
     private Stream stream = new Stream();
+    private boolean isContentAware = false;
+
+    public void setContentAwareness(StreamConfiguration streamConfiguration) {
+        List<StreamEntry> streamEntryList = streamConfiguration.getEntries();
+        for (StreamEntry streamEntry : streamEntryList) {
+            if ("$SOAPBody".equals(streamEntry.getValue())) { // Check if Dump Message Body was selected in UI
+                this.isContentAware = true;
+                return;
+            }
+        }
+        List<Property> properties = streamConfiguration.getProperties();
+        for (Property property : properties) {
+            if (property.isExpression() && property.getValue() != null) {
+                this.isContentAware = checkContentAware(property.getValue());
+            }
```

Comment:
Shouldn't we break the loop when content-aware property found ? Otherwise loop will continue and result will become the content-awareness of last property in the list